hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestListBlob.java

```diff
@@ -0,0 +1,140 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.fs.azurebfs;
+
+
+import java.util.List;
+
+import org.assertj.core.api.Assertions;
+import org.junit.Assume;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.azurebfs.services.AbfsClient;
+import org.apache.hadoop.fs.azurebfs.services.PrefixMode;
+import org.apache.hadoop.fs.azurebfs.utils.TracingContext;
+
+public class ITestListBlob extends
+    AbstractAbfsIntegrationTest {
+
+  public ITestListBlob() throws Exception {
+    super();
+  }
+
+  @Test
+  public void testListBlob() throws Exception {
+    AzureBlobFileSystem fs = getFileSystem();
+    assumeNonHnsAccountBlobEndpoint(fs);
+    int i = 0;
+    while (i < 10) {
+      fs.create(new Path("/dir/" + i));
+      i++;
+    }
+    List<BlobProperty> blobProperties = fs.getAbfsStore()
+        .getListBlobs(new Path("dir"), null,
+            Mockito.mock(TracingContext.class), null, null, false);
+    Assertions.assertThat(blobProperties)
+        .describedAs(
+            "BlobList should match the number of files created in tests + the directory itself")
+        .hasSize(11);
+
+    blobProperties = fs.getAbfsStore()
+        .getListBlobs(new Path("dir"), null,
+            Mockito.mock(TracingContext.class), null, null, true);
+    Assertions.assertThat(blobProperties)
+        .describedAs(
+            "BlobList should match the number of files created in tests")
+        .hasSize(10);
+  }
+
+  @Test
+  public void testListBlobWithMarkers() throws Exception {
+    AzureBlobFileSystem fs = getFileSystem();
+    assumeNonHnsAccountBlobEndpoint(fs);
+    int i = 0;
+    while (i < 10) {
+      fs.create(new Path("/dir/" + i));
+      i++;
+    }
```

Comment:
This test is covering the same as above , but has additional verifications. Can the above test be removed.

Reply:
The expectation of the above test is different. In the above test, we expect that there will be no markers that the server return. In the below tests, server will return markers.

Reply:
Have simplified the tests.

Reply:
In the older iteration, test scope were quite intermixed. In the new iteration:
1. testListBlob: test listBlob API with different number of maxResult.
2. testListBlobWithMarkers: test the usage of marker: here we will make server return only 1 object for a path for which 10 objects would be expected. When server returns object, it will also send the nextMarker, which client will use to make the call for remaining objects.